spyder_kernels/customize/spydercustomize.py

```diff
@@ -862,6 +834,43 @@ def get_debugger(filename):
     return debugger, filename
 
 
+def exec_code(code, filename, namespace, is_pdb):
+    """Execute code and display any exception."""
+    if PY2 and isinstance(filename, unicode):
+        filename = encode(filename)
+
+    ipython_shell = get_ipython()
+    try:
+        exec(compile(code, filename, 'exec'), namespace)
+    except SystemExit as status:
+        # ignore exit(0)
+        if status.code:
+            ipython_shell.showtraceback(exception_only=True)
+    except BaseException as error:
+        if isinstance(error, bdb.BdbQuit) and is_pdb:
+            # Ignore BdbQuit if we are debugging, as it is expected.
+            pass
+        else:
+            # We ignore the call to exec
+            ipython_shell.showtraceback(tb_offset=1)
+
+
+def get_file_code(filename):
+    """Retrive the content of a file."""
+    try:
+        # Get code from spyder
+        file_code = _frontend_request().get_file_code(filename)
+        if not PY2:
+            file_code = file_code.encode()
+        return file_code
+    except Exception:
+        mode = 'rb'
+        if PY2:
+            mode = 'r'
+        with open(filename, mode) as f:
+            return f.read()
```

Comment:
This is what is happening now. The kernel doesn't make any attempt to run a file from the remote machine, It just loads the local file with the given filename. Should that not be the case? If I write runfile(path), should the path be the path on the kernel or in the frontend machine?

Reply:
> If I write runfile(path), should the path be the path on the kernel or in the frontend machine?

Good question. We should maintain the same behavior as before for now, I think.

What do you say @bcolsen?

Reply:
But that means that you use spyder to run a file that you can not open with the editor.

Reply:
> What do we do with a working directory that doesn't exist currently?
Should we just not `os.chdir(wdir)`?

Reply:
> You can also now run files from the editor that can't be opened by the kernel.

> I think this implementation here is good. Remote kernel users are generally more advanced. I'm sure they will appreciate the limitations and new capabilities of the current implementation and a better one is on the way.

This makes more sense to me. The implementation now:

 - If the file is open in the editor, run this file. If the kernel is remote, that probably means that the working directory won't be set, but I am not sure how it could be. 

 - If the file is not open in the editor, load it from the kernel. If the kernel and the frontend are on the same machine, it shouldn't make a difference. If it is a remote machine, the user probably wants to load it from the kernel side?

Reply:
> Should we just not os.chdir(wdir)?

maybe print a warning?

Reply:
Should the `open(filename, mode)` fallback be on the frontend side or on the kernel side?

Reply:
As said in https://github.com/spyder-ide/spyder/pull/10115#discussion_r323080237, I think putting the fallback on the frontend side makes more sense. I will do that now.

Reply:
> > Should we just not os.chdir(wdir)?
> 
> maybe print a warning?

I think it might be best to put in an explanation as a notice whenever an editor script is run on a remote kernel. Like:
"Running file X from Spyder editor on a remote kernel. Working directory will not be set."

As for local new unsaved documents, I think we should also not set the working directory. We could have a similar notice.

Reply:
> I think it might be best to put in an explanation as a notice whenever an editor script is run on a remote kernel

I think this change doesn't belong to this PR, so please make the kernel to handle the case of a file not found in the editor for now.

Reply:
I will do that. The reason I thought it was relevant for this PR to take care of that was that it is about getting code from the frontend.